video.go

```diff
@@ -95,18 +95,19 @@ func (v *Video) extractDataFromPlayerResponse(prData playerResponseData) error {
 	v.Title = prData.VideoDetails.Title
 	v.Description = prData.VideoDetails.ShortDescription
 	v.Author = prData.VideoDetails.Author
-	v.Thumbnails = prData.VideoDetails.Thumbnail.Thumbnails
+	v.Thumbnails = prData.VideoDetails.Thumbnails
 
-	if seconds, _ := strconv.Atoi(prData.Microformat.PlayerMicroformatRenderer.LengthSeconds); seconds > 0 {
+	if seconds, _ := strconv.Atoi(prData.VideoDetails.LengthSeconds); seconds > 0 {
 		v.Duration = time.Duration(seconds) * time.Second
 	}
 
-	if str := prData.Microformat.PlayerMicroformatRenderer.PublishDate; str != "" {
-		v.PublishDate, _ = time.Parse(dateFormat, str)
-	}
+	// couldn't find any publishdate for now, have to add an another request
+	// if str := prData.VideoDetails.Microformat.PlayerMicroformatRenderer.PublishDate; str != "" {
+	// 	v.PublishDate, _ = time.Parse(dateFormat, str)
+	// }
 
 	// Assign Streams
-	v.Formats = append(prData.StreamingData.Formats, prData.StreamingData.AdaptiveFormats...)
+	v.Formats = append(prData.StreamingData.Formats, prData.StreamingData.Formats...)
```

Comment:
Why are you duplicating the `StreamingData.Formats` slice?

Reply:
https://github.com/cybre/youtube/commit/4a0c7d672485d96144fa26aebbfc602b3ad2bb2f

I forgot to change it from AdaptiveFormats to Format like you did in your solution